bindings/java/src/test/com/apple/foundationdb/test/AsyncStackTester.java

```diff
@@ -509,6 +509,37 @@ else if (op == StackOperation.TENANT_LIST_NAMES) {
 				inst.push(output);
 			}, FDB.DEFAULT_EXECUTOR);
 		}
+		else if (op == StackOperation.TENANT_LIST_METADATA) {
+			return inst.popParams(3).thenAcceptAsync(params -> {
+				byte[] begin = (byte[])params.get(0);
+				byte[] end = (byte[])params.get(1);
+				int limit = StackUtils.getInt(params.get(2));
+				CloseableAsyncIterator<KeyValue> tenantIter =
+				    TenantManagement.listTenants(inst.context.db, begin, end, limit);
+				ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
+				boolean validData = true;
+				try {
+					while (tenantIter.hasNext()) {
+						KeyValue next = tenantIter.next();
+						String metadata = new String(next.getValue());
+						// Without a JSON parsing library, we try to validate that the metadata consists
```

Comment:
If you wanted, you could probably extract this to somewhere like StackUtils.java and make it accessible to both testers.

Reply:
Are you referring to the entire try-catch block, the while loop, or the if-check?

Reply:
To the validation of the values